decode.go

```diff
@@ -1296,6 +1298,82 @@ func parseDurationNext(str string, totalNanos *big.Int) (string, error) {
 	return str, nil
 }
 
+// bytesUnitNames is the list of byte unit names.
+// E is exabyte, P is petabyte, T is terabyte, G is gigabyte, M is megabyte, K is kilobyte.
+// The i suffix indicates a binary unit (e.g., Ki = 1024).
+var bytesUnitNames = []string{"K", "M", "G", "T", "P", "E", "Ki", "Mi", "Gi", "Ti", "Pi", "Ei"}
+
+// A map from unit to the number of bytes in that unit.
+var bytesPerUnitMap = map[string]int64{
+	"K":  1000,
+	"Ki": 1 << 10,
+	"M":  1000 * 1000,
+	"Mi": 1 << 20,
+	"G":  1000 * 1000 * 1000,
+	"Gi": 1 << 30,
+	"T":  1000 * 1000 * 1000 * 1000,
+	"Ti": 1 << 40,
+	"P":  1000 * 1000 * 1000 * 1000 * 1000,
+	"Pi": 1 << 50,
+	"E":  1000 * 1000 * 1000 * 1000 * 1000 * 1000,
+	"Ei": 1 << 60,
+}
```

Comment:
k8s also supports `m` (milli): https://github.com/kubernetes/kubernetes/blob/ed9572d9c7733602de43979caf886fd4092a7b0f/staging/src/k8s.io/apimachinery/pkg/api/resource/suffix.go#L96-L97

This doesn't make sense for bytes but if we want to support CPU units this could be useful.